src/DotVVM.Framework/Routing/RouteBase.cs

```diff
@@ -186,5 +192,21 @@ public static void AddOrUpdateParameterCollection(IDictionary<string, object> ta
         /// </summary>
         public abstract IDotvvmPresenter GetPresenter(IServiceProvider provider);
 
+        private bool isFrozen = false;
+
+        private void ThrowIfFrozen()
+        {
+            if (isFrozen)
+                throw new InvalidOperationException($"The {this.GetType().Name} is frozen and can be no longer modified.");
+        }
+        public void Freeze()
+        {
+            this.isFrozen = true;
+            FreezableDictionary.Freeze(ref this._defaultValues);
+
+            Freeze2();
+        }
+
+        protected abstract void Freeze2();
```

Comment:
Come on, `Freeze2`? What kind of a name is that. Is there a problem with making `Freeze` virtual?

Reply:
The problem is that it would allow you to override the `Freeze` method in a derived class, which would allow you to suppress the freeze. I don't care about the name of `Freeze2`, maybe we could also name it `Freeze_IHATEINHERITANCE`

Reply:
@exyi  Please, name it OnFrozen(). We can at least use .net conventions. When it's called OnFrozen you know that method is invoked after configuration is frozen. Freeze2 is nothing-saying.

Reply:
Or maybe the better name is even OnFreezing()

Reply:
`OnFreezing` or `FreezeCore` looks better than `Freeze2`.

Reply:
`FreezeCore` is a bad name, because it's not a core of freeze.

`OnFreezing` is bad name, because it's not an event you can optionally handle - it's a method you must implement and make sure that all properties are frozen.

I think that `Freeze2` will trigger a potential implementer of a derived class, which increases the probability it will get implemented correctly.

Also, if the implementer would want to allow inheritance from his class, he'd have to create another `Freeze` method for them. That would be named `Freeze3`, which is much better than `OnOningFreezing` or `TheRealCoreOfFreeze` :)

Reply:
Still, I don't like it. But now I understand your POV.